modules/core/src/test/java/io/fluo/core/impl/CollisionIT.java

```diff
@@ -0,0 +1,162 @@
+      tx.mutate().row(num).fam("stat").qual("changed").weaklyNotify();
+    }
+  }
+
+  public static class TotalObserver extends TypedObserver {
+
+    @Override
+    public ObservedColumn getObservedColumn() {
+      return new ObservedColumn(typeLayer.bc().fam("stat").qual("changed").vis(), NotificationType.WEAK);
+    }
+
+    @Override
+    public void process(TypedTransactionBase tx, Bytes row, Column col) {
+      int total = tx.get().row(row).fam("stat").qual("total").toInteger();
+      int processed = tx.get().row(row).fam("stat").qual("processed").toInteger(0);
+
+      tx.mutate().row(row).fam("stat").qual("processed").set(total);
+      tx.mutate().row("all").fam("stat").qual("total").increment(total - processed);
+    }
+  }
+
+  @Override
+  protected List<ObserverConfiguration> getObservers() {
+    return Arrays.asList(new ObserverConfiguration(TotalObserver.class.getName()));
+  }
+
+  @Override
+  protected void setConfig(FluoConfiguration config) {
+    config.setLoaderQueueSize(20);
+    config.setLoaderThreads(20);
+    config.setWorkerThreads(20);
+
+    // make updates in ZK related to Fluo GC more frequent
+    config.setProperty(ZookeeperUtil.ZK_UPDATE_PERIOD_PROP, "100");
+  }
+
+  @Test
+  public void testLotsOfCollisions() throws Exception {
+
+    Random rand = new Random(45734985);
+
+    int[] nums = new int[1000];
+    int[] expectedCounts = new int[5];
+
+    for (int i = 0; i < nums.length; i++) {
+      nums[i] = rand.nextInt(expectedCounts.length);
+      expectedCounts[nums[i]]++;
+    }
+
+    try (LoaderExecutor loader = client.newLoaderExecutor()) {
+      for (int num : nums) {
+        loader.execute(new NumLoader(num));
+      }
+    }
+
+    miniFluo.waitForObservers();
+
+    try (TypedSnapshot snapshot = typeLayer.wrap(client.newSnapshot())) {
+
+      for (int i = 0; i < expectedCounts.length; i++) {
+        Assert.assertEquals(expectedCounts[i], snapshot.get().row(i).fam("stat").qual("total").toInteger(-1));
+      }
+
+      Assert.assertEquals(1000, snapshot.get().row("all").fam("stat").qual("total").toInteger(-1));
+    }
+
+    long oldestTS = ZookeeperUtil.getOldestTimestamp(config.getZookeepers());
+    while (true) {
+      UtilWaitThread.sleep(300);
+      long tmp = ZookeeperUtil.getOldestTimestamp(config.getZookeepers());
+      if (oldestTS == tmp) {
+        break;
+      }
+      oldestTS = tmp;
+    }
+
+    conn.tableOperations().compact(getCurTableName(), null, null, true, true);
+
+    Scanner scanner = conn.createScanner(getCurTableName(), Authorizations.EMPTY);
+
+    HashSet<String> rowCols = new HashSet<>();
+
+    for (Entry<Key,Value> entry : scanner) {
+      Key k = entry.getKey();
+      String rowCol = k.getRow() + ":" + k.getColumnFamily() + ":" + k.getColumnQualifier() + ":"
+          + String.format("%x", k.getTimestamp() & ColumnConstants.PREFIX_MASK);
+      Assert.assertFalse("Duplicate row col " + rowCol, rowCols.contains(rowCol));
```

Comment:
If you change the test and add calls to `printTable()` (defined in `ITBase`) before and after compaction, should see the duplicates